coriolis/osmorphing/base.py

```diff
@@ -393,39 +394,78 @@ def _set_selinux_autorelabel(self):
-        for cloud_cfg_path in cloud_cfg_paths:
-            if not self._test_path(cloud_cfg_path):
-                LOG.warn("Could not find %s. Skipping reconfiguration." % (
-                    cloud_cfg_path))
-                continue
+            "Configuring cloud-init additional options")
+        new_cloud_cfg = yaml.dump(cloud_cfg)
+        self._write_file_sudo(cloud_config_path, new_cloud_cfg)
+
+    def _disable_installer_cloud_config(self):
+        installer_config_path = "/etc/cloud/cloud.cfg.d/99-installer.cfg"
+        if self._test_path(installer_config_path):
+            self._event_manager.progress_update(
+                "Disabling Ubuntu installer-generated cloud-config")
+            self._exec_cmd_chroot(
+                f"mv {installer_config_path} {installer_config_path}.bak")
+
+    def _ensure_cloud_init_not_disabled(self):
+        disabler_file = "/etc/cloud/cloud-init.disabled"
+        system_conf_disabler = "/etc/systemd/system.conf"
+        grub_conf_disabler = "/etc/default/grub"
+        if self._test_path(disabler_file):
+            self._exec_cmd_chroot(f"rm {disabler_file}")
+        if self._test_path(system_conf_disabler):
+            self._exec_cmd_chroot(
+                "sed -i '/cloud-init=disabled/d' %s" % system_conf_disabler)
+        if self._test_path(grub_conf_disabler):
+            self._exec_cmd_chroot(
+                "sed -i '/cloud-init=disabled/d' %s" % grub_conf_disabler)
+
+    def _reset_cloud_init_run(self):
+        self._exec_cmd_chroot("cloud-init clean --logs")
+
+    def _get_default_cloud_user(self):
+        cloud_cfg_path = 'etc/cloud/cloud.cfg'
+        if not self._test_path(cloud_cfg_path):
+            return DEFAULT_CLOUD_USER
+        cloud_cfg_content = self._read_file_sudo(cloud_cfg_path)
+        cloud_cfg = yaml.load(cloud_cfg_content, Loader=yaml.SafeLoader)
+        return cloud_cfg.get('system_info', {}).get('default_user', {}).get(
+            'name', DEFAULT_CLOUD_USER)
+
+    def _create_cloudinit_user(self):
+        cloud_user = self._get_default_cloud_user()
+        if not self._check_user_exists(cloud_user):
+            self._exec_cmd_chroot("useradd %s" % cloud_user)
+
+    def _configure_cloud_init(self):
+        cloud_cfg_mods = {}
+        if "cloud-init" not in self.get_packages()[0]:
+            return
+
+        self._disable_installer_cloud_config()
+        self._ensure_cloud_init_not_disabled()
+        self._reset_cloud_init_run()
+
+        if self._osmorphing_parameters.get('retain_user_credentials', False):
+            cloud_cfg_user_retention = {
+                'disable_root': False,
+                'ssh_pwauth': True,
+                'users': None
+            }
+            cloud_cfg_mods.update(cloud_cfg_user_retention)
+        else:
+            self._create_cloudinit_user()
+
+        if not self._osmorphing_parameters.get('set_dhcp', False):
```

Comment:
Please set `set_dhcp`'s default to True. Otherwise, in providers where we do not support this option, it will always disable cloud-init networking. 
```suggestion
        if not self._osmorphing_parameters.get('set_dhcp', True):
```